### PR TITLE
fix(trace) account for head or tail end of icons

### PR DESCRIPTION
**Pull request by JonasBa**
Span bars sometimes have icons at the head or tail end, which means they can overlap with the duration bars and cause visual issues. This PR computes the span bar space according to the min and max timestamps so that the duration labels no longer overlap.

Before:

https://github.com/getsentry/sentry/assets/9317857/d8ff6efc-7145-49ff-a7e1-4ab6a8c21f0a

After:

https://github.com/getsentry/sentry/assets/9317857/fd23c752-e5b3-4623-885f-e51460074a75

### CODE REVIEW

File: static/app/views/performance/newTraceDetails/traceRenderers/virtualizedViewManager.tsx [hunk 1]
```diff
@@ -1170,57 +1170,28 @@ export class VirtualizedViewManager {
...
-
-    // |   text|
+    const right_outside = this.computeTransformXFromTimestamp(text_right) + TEXT_PADDING;
+    // |---text|
     const right_inside =
-      this.computeTransformXFromTimestamp(span_space[0] + span_space[1]) -
-      width -
-      TEXT_PADDING;
-    // |text   |
+      this.computeTransformXFromTimestamp(span_space[0] + span_space[1]) - TEXT_PADDING;
```
Comment (Zylphrex) [inline static/app/views/performance/newTraceDetails/traceRenderers/virtualizedViewManager.tsx:1189]: Did this need to subtract `width` as well?

File: static/app/views/performance/newTraceDetails/traceRenderers/virtualizedViewManager.tsx [hunk 2]
```diff
@@ -1704,6 +1675,50 @@ export class VirtualizedViewManager {
...
+  node: TraceTreeNode<any>,
+  span_space: [number, number],
+  icon_width: number
+) {
+  if (!node.errors.size && !node.performance_issues.size) {
+    return [span_space[0], span_space[0] + span_space[1]];
+  }
+
+  let min_icon_timestamp = span_space[0];
+  let max_icon_timestamp = span_space[0] + span_space[1];
```
Comment (Zylphrex) [inline static/app/views/performance/newTraceDetails/traceRenderers/virtualizedViewManager.tsx:1691]: Nit: Just makes what you're returning a little more obvious

```suggestion
  let min_icon_timestamp = span_space[0];
  let max_icon_timestamp = span_space[0] + span_space[1];
  
    if (!node.errors.size && !node.performance_issues.size) {
    return [min_icon_timestamp, max_icon_timestamp];
  }
```

**Review by Zylphrex — APPROVED** [review]
There's a mix of snake case and camel case here that you may want to clean up at some point.

File: static/app/views/performance/newTraceDetails/traceRenderers/virtualizedViewManager.tsx (hunk 1 above)
Comment (JonasBa) [reply static/app/views/performance/newTraceDetails/traceRenderers/virtualizedViewManager.tsx]: Yup, and the padding needs to be negative here too.

**Pull request merged**